standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/parser/ExpressionTree.java

```diff
@@ -18,7 +18,7 @@
 package org.apache.hadoop.hive.metastore.parser;
 
 import java.sql.Timestamp;
-import java.util.Date;
```

Comment:
That's a sneaky change since `java.util.Date` is a super class of `java.sql.Date` so the behavior of `instanceof` and other operators may be affected. 

Do we have test coverage for this class? Are we creating anywhere else instances of `java.util.Date` that may end up here?

Reply:
Actually in the flow `java.sql.Date `object only coming here, initially when added it was added wrongly as `java.util.Date`. As `java.util.Date` is super class for `java.sql.Date`, `value instanceof java.util.Date` is working fine till now.   `partition_date.q` test is covering it.